addons/debugger/userscript.js

```diff
@@ -0,0 +1,255 @@
+export default async function ({ addon, global, console, msg }) {
+  let workspace, showingConsole;
+  const img = document.createElement("img");
+  img.className = `debug-btn ${addon.tab.scratchClass("button_outlined-button")}`;
+  img.src = addon.self.dir + "/debug.svg";
+  img.draggable = false;
+  img.title = msg("debug");
+  img.addEventListener("click", () => toggleConsole());
+
+  const vm = addon.tab.traps.vm;
+  addon.tab.addBlock("sa-log %s", ["content"], ({ content }, targetId, blockId) => {
+    workspace = Blockly.getMainWorkspace();
+    addItem(content, targetId, blockId, "log");
+  });
+  addon.tab.addBlock("sa-warn %s", ["content"], ({ content }, targetId, blockId) => {
+    workspace = Blockly.getMainWorkspace();
+    addItem(content, targetId, blockId, "warn");
+  });
+  addon.tab.addBlock("sa-error %s", ["content"], ({ content }, targetId, blockId) => {
+    workspace = Blockly.getMainWorkspace();
```

Comment:
why don't you get the workspace inside of the `addItem` function (to avoid duplication)

Reply:
Because idk. It was causing weird bugs.